src/pybamm/util.py

```diff
@@ -108,38 +109,100 @@ def __getitem__(self, key):
                 f"'{key}' not found. Best matches are {best_matches}"
             ) from error
 
-    def search(self, key, print_values=False):
+    def _find_matches(self, search_key, known_keys):
```

Comment:
While we are at it :)
```suggestion
    def _find_matches(self, search_key: str, known_keys: list[str]):
```